google_guest_agent/network/manager/common.go

```diff
@@ -19,13 +19,37 @@ package manager
 import (
 	"fmt"
 	"net"
+	"os/exec"
 
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/ps"
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
 	"github.com/GoogleCloudPlatform/guest-agent/metadata"
 	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
 )
 
+const (
+	// Values for determining the behavior of mocked functions in testing.
+	valueFalse = iota
+	valueTrue
+	valueErr
+)
+
 var (
 	badMAC = make(map[string]net.Interface)
+
+	// The following are set for testing purposes.
+	// runQuiet points to the function to run commands without output.
+	runQuiet = run.Quiet
```

Comment:
I would suggest looking at [MDS interface](https://github.com/GoogleCloudPlatform/guest-agent/blob/main/metadata/metadata.go#L53) how its implemented and used for unit testing and see if it makes sense introducing something like that for `ps` or `run`. Problem I see with doing like this is every other test that needs to mock `ps` or `exec` might've to come up with variables like `findProcess` here v/s having consistent interface implementation and just mocking behavior in tests.

Reply:
Created an interface and an overrideable exported variable for overriding the client with a test/mock implementaiton.